openfe/setup/molecule.py

```diff
@@ -28,8 +59,9 @@ class Molecule:
         will be used in the hash.
     """
     def __init__(self, rdkit: RDKitMol, name: str = ""):
-        # internally we store as RDKit, for now...
-        self._rdkit = Chem.Mol(rdkit)
+        name = _ensure_ofe_name(rdkit, name)
+        _ensure_ofe_version(rdkit)
+        self._rdkit = rdkit
```

Comment:
This has removed the copy on the input, which might be required, I'll let you put it back in if you want later